base/linalg/triangular.jl

```diff
@@ -324,20 +325,21 @@ function scale!{T<:Union{LowerTriangular, UnitLowerTriangular}}(A::LowerTriangul
             @inbounds A[i,j] = c * B[i,j]
         end
     end
+    return A
 end
+scale!(A::Union{UpperTriangular,LowerTriangular},c::Number) = scale!(A,A,c)
+scale!(c::Number, A::Union{UpperTriangular,LowerTriangular}) = scale!(A,c)
 
 for (t1, t2) in ([:UnitLowerTriangular, :LowerTriangular],
                  [:UnitUpperTriangular, :UpperTriangular])
-    @eval begin
-        function scale!{T,S<:Number}(A::$(t1){T}, c::S)
-            D = $(t2)(Array(promote_type(T,S), size(A)...));
-            scale!(D,A,c);
-            return D
-        end
-        function scale!{T,S<:Number}(A::$(t2){T}, c::S)
-            D = $(t2)(Array(promote_type(T,S), size(A)...));
-            scale!(D,A,c);
-            return D
+    for (type1, type2) in ([:Real, :Complex], [:Any, :Number])
```

Comment:
Because if I don't put the `Real`/`Complex` version I get a few warnings at compile time, pointing out that my methods are ambiguous with the `scale(A,c)` in `generic.jl`
```javascript
WARNING: New definition 
    scale(Base.LinAlg.UnitLowerTriangular{#T<:Any, S<:AbstractArray{T<:Any, 2}}, #S<:Number) at linalg/triangular.jl:337
is ambiguous with: 
    scale(AbstractArray{#R<:Real, N<:Any}, #S<:Base.Complex) at linalg/generic.jl:9.
To fix, define 
    scale(Base.LinAlg.UnitLowerTriangular{_<:Real, S<:AbstractArray{T<:Any, 2}}, #S<:Base.Complex)
before the new definition.
WARNING: New definition 
    scale(#S<:Number, Base.LinAlg.UnitLowerTriangular) at linalg/triangular.jl:341
is ambiguous with: 
    scale(Base.Complex, AbstractArray{#R<:Real, N<:Any}) at linalg/generic.jl:14.
To fix, define 
    scale(_<:Base.Complex, Base.LinAlg.UnitLowerTriangular{#R<:Real, _<:S<:AbstractArray{T<:Any, 2}})
before the new definition.
WARNING: New definition 
    scale(Base.LinAlg.UnitUpperTriangular{#T<:Any, S<:AbstractArray{T<:Any, 2}}, #S<:Number) at linalg/triangular.jl:337
is ambiguous with: 
    scale(AbstractArray{#R<:Real, N<:Any}, #S<:Base.Complex) at linalg/generic.jl:9.
To fix, define 
    scale(Base.LinAlg.UnitUpperTriangular{_<:Real, S<:AbstractArray{T<:Any, 2}}, #S<:Base.Complex)
before the new definition.
WARNING: New definition 
    scale(#S<:Number, Base.LinAlg.UnitUpperTriangular) at linalg/triangular.jl:341
is ambiguous with: 
    scale(Base.Complex, AbstractArray{#R<:Real, N<:Any}) at linalg/generic.jl:14.
To fix, define 
    scale(_<:Base.Complex, Base.LinAlg.UnitUpperTriangular{#R<:Real, _<:S<:AbstractArray{T<:Any, 2}})
before the new definition.
```